Add conflict rules to avoid installing both the snoopy and libsnoopy pkgs

fredericmora opened the pull request:
The snoopy package is available on the Ubuntu repos and should be replaced if libsnoopy is installed, especially if a new feature is configured in the snoopy.ini file.

bostjan commented:
True.

BTW how do you set versions here, what guides the decision? I see now things changed to 2.3.1rc1, which is a version of Snoopy that currently does not exist. Should we do something about it, for example should ./configure / make set variables in this file to correct version?

b.

bostjan commented:
Also, how do you decide whether package should be called 'snoopy' or 'libsnoopy'?

Just curious, as I have not created any Ubuntu packages myself.

b.

fredericmora commented:
Regarding version numbers: I am trying to follow the semantic versioning principle. In a version number x.y.z, x is bumped up indicates major rewrites and incompatibilities with the previous version. Incrementing y indicates a new feature or minor change (which is why I bumped that value for the new filter feature). Incrementing z indicates a packaging or distribution change.

Since I changed the packaging only, I bumped z. I also try to follow your rc_n_ numbering convention, so I reset the rc counter to 1 since I changed x.y.z.

Now, since this is a bugfix, you are perfectly entitled to make the version number decision. You may decide that this 2.3.1rc1, since it's unreleased, should actually be the next RC for your current version (2.3.0rc2). That'd be perfectly acceptable. The numbers are not cast in stone until you release.

fredericmora commented:
Regarding the binary package name: Both names are acceptable. The existing Ubuntu package is called "snoopy" and provides /lib/snoopy.so. So integration in existing boxes would be smoother if you named your binary "snoopy".

The only purpose of this bugfix is to make your libsnoopy take over the snoopy files and kick out the snoopy package if it's already installed. So of course, if you decide to do rename your package to "snoopy", you can totally disregard this bugfix. I wouldn't mind at all, same result.

bostjan commented:
Seems reasonable. I do not have a strong opinion about ubuntu packaging versioning one way or the other, therefore a non-issue.

b.
